### ISSUE TITLE
Ballerina shell should allow `public`

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/ballerina-lang#28283]
Ballerina shell gives me an error for e.g.

```
public int x = 0;
```

I am guessing the thinking was that `public` is pointless, because there is no way for another module to reference symbols. But often people will want to use the shell to try out snippets of code that will be incorporated into programs, and these snippets may well have `public `in them.

**Comment by kdsuneraavinash** [comment]
This seems to be unsupported even in the compiler yet. For example following fails to compile, (Ballerina Swan Lake Alpha 1)
```ballerina
public int i = 100;

public function main() {
}
```
However, because defining `public` in the shell is pointless, maybe allowing public in the shell, but silently removing `public` qualifier would be a solution?

**Comment by jclark** [comment]
It looks like the compiler hasn't implemented `public` variables yet: #27855 (module-level variables used not to be allowed to be public).

But the following doesn't work in the shell, but does work outside the shell:

```
public function foo() {
}
```

I get:

```
| Invalid statement: Invalid qualifier 'public'. Public is not allowed here.
| Parsing aborted because of errors.
```

**Comment by kdsuneraavinash** [comment]
Previously `public` qualifier was altogether rejected from every module level dcln.
Should be fixed by #28319 
However, even after that fix, variable declarations will cause compilation errors on public var dclns. But those will be fixed once compiler supports public var dclns.
